File: Back-End-SpringBoot/src/main/java/com/wastedTimeStudiyng/service/StudentService.java
package com.wastedTimeStudiyng.service;

import java.util.List;

import com.wastedTimeStudiyng.pojos.Student;

public interface StudentService {

	//CRUD
	public List<Student> getStudents();
	public Student getStudentById(int id);
	public void createStudent(Student student);
	public void updateStudent(String dni, Student student);
	public void deleteStudent(String dni);
	
	//Other methods
	public Student getStudentByDni(String dni);
	
}

File: Front-End-Angular/src/app/services/student.service.ts
import { Injectable } from '@angular/core';

import { Observable, of } from 'rxjs';
import { Student } from '../components/student/student';
import { ClientWSService } from '../webservice/clientWS/client-ws.service'


@Injectable({
  providedIn: 'root'
})
export class StudentService {

  constructor(private clientWSService: ClientWSService) { }
  
  //When you send a http request he returned a Observable when transform the JSON response from WS(Web Service)
  public getStudents(): Observable<Student[]> {
    return this.clientWSService.getStudents();
  }

  public getStudentById(id : number): Observable<Student>{
    return this.clientWSService.getStudentById(id);
  }

  public deleteStudent(dni: String): void {
    this.clientWSService.deleteStudent(dni).subscribe();
  }
  
}
